src/Middleware/HeaderPropagation/src/DependencyInjection/HeaderPropagationExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Http;
+using Microsoft.Extensions.Http.HeaderPropagation;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class HeaderPropagationExtensions
+    {
+        public static IServiceCollection AddHeaderPropagation(this IServiceCollection services, Action<HeaderPropagationOptions> configure)
+        {
+            services.AddHttpContextAccessor();
```

Comment:
IHttpContextAccessor shouldn't be needed of you're going to properly avoid the concurrency issue.